src/logging.c

```diff
@@ -72,6 +103,44 @@ void rcutils_log(
   }
 }
 
+/*!
+  RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER(
+   n, output_buffer_size, allocator, output_buffer, static_output_buffer)
+  Macro for (re)allocating a dynamic buffer if the ouput buffer doesn't have enough space for an
+  additional n characters and null terminator. Whether to allocate or re-allocate is determined by
+  if the output_buffer points to the original static_output_buffer or not. If a new buffer is
+  allocated it will be padded with NUL.
+*/
+
+#define RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER( \
```

Comment:
Pay attention for bad practices here because before this I didn't have much experience writing macros like this.

Reply:
If we were to switch to a function I think the readability of the code would stay the same, wouldn't it? (we'd remove trailing `\`, use pointers for buffer sizes, but otherwise the code would stay the same). The duplication of code at each of the callers (e.g. error checking) makes me think we wouldn't gain much from switching, but I might have missed something

Reply:
In general, macros are difficult to read and reason about, and difficult to debug in a debugger later.  As an instance of the difficulty of reasoning about it, the fact that the macro accesses the `size` variable, which is in the "outer scope" of the generated code, means that the reader has to go find size in the correct scope to know what is going on with it (this is similar to using global variables in some respects).  If we changed it to a function I don't think the code size would change very much, but I think it would be a lot easier to read and reason about.

All of that being said, this seems to be a common pattern in this codebase.  If others are OK with the macros, then we can go ahead with it.

Reply:
ok, I understand your point better now when considering the reasoning about the code. I still don't know that too much would change for this particular case due to the nature of the arguments (e.g. `output_buffer_size`) being both inputs and outputs, but I'll keep your point about "reasonability" in mind for the future when making the decision for one or the other